src/js/components/NewsFeedPost.jsx

```diff
@@ -1,59 +1,67 @@
-import React from "react";
-import NewsFeedStore from "stores/NewsFeedStore";
+import React from 'react';
+import NewsFeedStore from 'stores/NewsFeedStore';
 import FaHeart from 'react-icons/lib/fa/heart';
 import FaRetweet from 'react-icons/lib/fa/retweet';
 
 export default class NewsFeedPost extends React.Component {
 
   constructor(props) {
     super(props);
-    // this.togglePictureDisplay = this.togglePictureDisplay.bind(this);
-    this.changeLike = this.changeLike.bind(this);
     this.state = this.props;
+    this.changeLike = this.changeLike.bind(this);
+    this.retweet = this.retweet.bind(this);
+    this.showImage = this.showImage.bind(this);
   }
 
-  // componentWillMount() {
-  //   NewsFeedStore.on("updated", this.getNewsFeedPosts);
-  //   NewsFeedStore.on("error", this.showError);
-  // }
+  // componentWillMount() {}
 
-  // componentWillUnmount() {
-  //   NewsFeedStore.removeListener("updated", this.getNewsFeedPosts);
-  //   NewsFeedStore.removeListener("error", this.showError);
-  // }
-
-  showError() {
-    console.log(NewsFeedStoreStore.error);
-  }
+  // componentWillUnmount() {}
 
   changeLike() {
     if (this.state.current_user_likes) {
-      NewsFeedStore.decrementLike();
+      NewsFeedStore.unlikePost(this.state.id);
+      // TODO: Potentially move this logic to be based on emitted action?
       this.setState({
         current_user_likes: false,
+        like_count: (parseInt(this.state.like_count) - 1).toString(),
       });
     } else {
-      NewsFeedStore.incrementLike();
+      NewsFeedStore.likePost(this.state.id);
       this.setState({
         current_user_likes: true,
+        like_count: (parseInt(this.state.like_count) + 1).toString(),
       });
     }
+  }
 
+  retweet() {
+    const retweetText = '"' + this.state.text + '" - ' + this.state.poster_name + '\n';
```

Comment:
Could use string interpolation to make this a bit easier (wrap in `).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals